src/main/scala/com/cloudera/sparkts/ARIMA.scala

```
@@ -0,0 +1,566 @@
+   * the autoregression terms, d represents the order of differencing, and q moving average error
+   * terms. If includeIntercept is true, the model is fitted with a mean. The user can specify
+   * the method used to fit the parameters. The current implementation only supports "css",
+   * meaning conditional sum of squares. In order to select the appropriate order of the model,
+   * users are advised to visualize inspect ACF and PACF plots. Finally, current implementation
+   * does not verify if the parameters fitted are stationary/invertible. Given this, it is suggested
+   * that users inspect the resulting model. However, if parameters are not sane, it is likely
+   * that the order of the model fit is inappropriate.
+   * We follow the specfication outlined in
+   * https://www.otexts.org/fpp/8/5
+   */
+  def fitModel(
+      order:(Int, Int, Int),
+      ts: Vector[Double],
+      includeIntercept: Boolean = true,
+      method: String = "css-BOBYQA"): ARIMAModel = {
+    val (p, d, q) = order
+    val diffedTs = differences(ts, d).toArray.drop(d)
+
+    if (p > 0 && q == 0) {
+      val arModel = Autoregression.fitModel(new DenseVector(diffedTs), p)
+      return new ARIMAModel(order, Array(arModel.c) ++ arModel.coefficients, includeIntercept)
+    }
+
+    // Initial parameter guesses
+    val initParams = HannanRisannenInit(diffedTs, order, includeIntercept)
+
+    // TODO: Enforce stationarity and invertibility for AR and MA terms
+    method match {
+      case "css-BOBYQA" => {
+        val params = fitWithCSSBOBYQA(diffedTs, order, includeIntercept, initParams)
+        new ARIMAModel(order, params, includeIntercept)
+      }
+      case "css-CGD" => {
+        val params = fitWithCSSCGD(diffedTs, order, includeIntercept, initParams)
+        new ARIMAModel(order, params, includeIntercept)
+      }
+      case "ml" => throw new UnsupportedOperationException()
```

Should throw an UnsupportedOperationException on any input that's not one of the two css methods